### DISCUSSION TITLE
Set up MQTT to report in degF, homebridge thinks its degC

### DISCUSSION THREAD

**Discussion by agprimatic** [aqualinkd/AqualinkD#205] [General]
I have AqualinkD configured to report the temperature in degF (using convert_mqtt_temp_to_c=no).
The homebridge-aqualinkd plugin thinks the temp is in degC, so it converts it (again) to degF.
Would be nice to either have this communicated between AqualinkD and the plugin, or provide an option to use degF in the plugin.

**Comment by sfeakes** [comment]
It is documented in the config & wiki that you have to convert to degC for HomeKit.   HomeKit expects degC and that can't be changed, This is also the same for many other home automation hubs.
The only way around this would be to post multiple MQTT topics, but the overhead of that is worse than simply converting on the MQTT consumer.
On that note, Why can't this be converted on the MQTT consumer?

**Comment by johntdyer** [comment]
How exactly do you convert this on the consumer ?

**Comment by sfeakes** [comment]
That depends on the consumer.  But the algorithm is ( degC * 1.8 + 32 )   Where degC is the value received from the MQTT message.

Things like home assistant you simply tell it the unit of measure is degC.

**Comment by agprimatic** [comment]
OK. The only wrinkles are that my home control processor doesn't support floating point, which makes the conversion of degC integers to degF integers imprecise. Or it requires that I keep both the integer and fractional values for every temperature so the conversions are lossless.

And the aqualink temperature increment and decrement would now be in units of degC, which will be a much bigger jump than the one degF expected on the user interface.

Would be much easier for my application if you had both degC and degF available over MQTT, and allowed whichever temperature unit defined in the .conf file to be the controlling unit of temperature (with the other just being a conversion).
